### ISSUE TITLE
Details view not working anymore

### ISSUE THREAD

**Issue by mkansy** [vivekratnavel/omniboard#207]
Hi, 

First of all, I just started using sacred and Omniboard yesterday, so it could be an issue on my part. Also, thanks for making this tool and offering support!

I set up a free MongoDB Atlas cluster yesterday, followed a tutorial to install Node.js (12.16.1), and then installed the newest version of Omniboard (2.8.1?) according to the quick start guide. Everything seemed to work fine, and I could see my sample runs on Omniboard. After a few more hours of starting runs with sacred and adding a few sample metrics, Omniboard stopped working for some reason. When I open it, it still shows me all of the runs with the configuration parameters as columns. When I click on the arrow to show details, however, all fields are empty (including the configuration parameters, see screenshot below). When I keep pressing the different fields in the details view, eventually I just get a blank screen.

![Omniboard](https://user-images.githubusercontent.com/50331699/78346001-b75f9a80-759e-11ea-9af5-a5789c4263cb.png)

I tried the following things to fix this issue without success:
- Resetting cache
- Checking whether the data is even stored in the database
- Using a different browser
- Using a different computer
- Creating and using a new MongoDB Atlas cluster
- Using a different Omniboard version (2.7.3)

Do you have any idea what could be causing this problem? Let me know if you need any more information from me. Thanks!

**Comment by vivekratnavel** [comment]
Can you post omniboard server logs and the network tab in Browser dev console?

https://developers.google.com/web/tools/chrome-devtools/open

**Comment by mkansy** [comment]
I just started Omniboard and then pressed on the detail arrow to reproduce the error. I think the auto-refresh is also in the following logs once or twice.

The Omniboard server logs are here (Note: I replaced my username, password, and the cluster name with \<username\>, \<password\>, and \<cluster\>):
`{
  _: [],
  mu: 'mongodb+srv://<username>:<password>@<cluster>.mongodb.net/thesis_runs?retryWrites=true',
  '$0': '/usr/local/bin/omniboard'
}
Omniboard is listening on port 9000!
(node:5475) DeprecationWarning: current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
Connection to mongodb+srv://<username>:<password>@<cluster>.mongodb.net/thesis_runs?retryWrites=true established successfully!
::1 - - [04/Apr/2020:08:51:13 +0000] "GET /sacred HTTP/1.1" 304 - 4.839 ms
::1 - - [04/Apr/2020:08:51:14 +0000] "GET /sacred/api/v1/database HTTP/1.1" 304 - 0.476 ms
::1 - - [04/Apr/2020:08:51:14 +0000] "GET /sacred/api/v1/Version HTTP/1.1" 304 - 0.405 ms
::1 - - [04/Apr/2020:08:51:14 +0000] "GET /sacred/api/v1/Omniboard.Settings HTTP/1.1" 304 - 205.197 ms
::1 - - [04/Apr/2020:08:51:14 +0000] "GET /sacred/api/v1/Omniboard.Metric.Columns HTTP/1.1" 304 - 19.103 ms
::1 - - [04/Apr/2020:08:51:15 +0000] "GET /sacred/api/v1/Omniboard.Custom.Columns HTTP/1.1" 304 - 370.301 ms
::1 - - [04/Apr/2020:08:51:15 +0000] "GET /sacred/api/v1/Runs?distinct=omniboard.tags HTTP/1.1" 304 - 31.719 ms
Query took 0.276 seconds.
::1 - - [04/Apr/2020:08:51:15 +0000] "GET /sacred/api/v1/Runs?select=_id,experiment.name,host.hostname,format,command,start_time,status,omniboard.tags,omniboard.notes,resources,heartbeat,duration,result,stop_time,config.mongo_url,config.seed,config.tags,meta.comment&sort_by=_id&order_by=-1&query=%7B%7D&limit=50 HTTP/1.1" 304 - 276.962 ms
Query took 0.291 seconds.
::1 - - [04/Apr/2020:08:51:15 +0000] "GET /sacred/api/v1/Runs/count?query=%7B%7D HTTP/1.1" 304 - 291.209 ms
::1 - - [04/Apr/2020:08:51:21 +0000] "GET /sacred/api/v1/Metrics?query=%7B%22run_id%22:%221%22%7D HTTP/1.1" 304 - 26.781 ms
Query took 0.038 seconds.
::1 - - [04/Apr/2020:08:51:21 +0000] "GET /sacred/api/v1/Runs/1?select=captured_out,info,meta,host,experiment,artifacts,config,fail_trace,status HTTP/1.1" 304 - 38.915 ms
Query took 0.05 seconds.
::1 - - [04/Apr/2020:08:51:44 +0000] "GET /sacred/api/v1/Runs/count?select=_id,experiment.name,host.hostname,format,command,start_time,status,omniboard.tags,omniboard.notes,resources,heartbeat,duration,result,stop_time,config.mongo_url,config.seed,config.tags,meta.comment&sort_by=_id&order_by=-1&query=%7B%7D&limit=50 HTTP/1.1" 304 - 50.214 ms
Query took 0.352 seconds.
::1 - - [04/Apr/2020:08:52:15 +0000] "GET /sacred/api/v1/Runs/count?select=_id,experiment.name,host.hostname,format,command,start_time,status,omniboard.tags,omniboard.notes,resources,heartbeat,duration,result,stop_time,config.mongo_url,config.seed,config.tags,meta.comment&sort_by=_id&order_by=-1&query=%7B%7D&limit=50 HTTP/1.1" 304 - 353.008 ms`

A screenshot of the network tab is here:
![Omniboard_Network](https://user-images.githubusercontent.com/50331699/78423062-028fb100-7664-11ea-9069-c370f19e8f0c.png)

Thanks for your help!

**Comment by vivekratnavel** [comment]
Hi, 

I am unable to reproduce this issue on my end. 

But, I will try to debug it for you. Can you post the response of the call `1?select=captured_out,...`? Also please post the `Request URL` from `Headers` tab. Thanks!

**Comment by mkansy** [comment]
Hi,

The response is:
`[]`

The Request URL is:
`http://localhost:9000/sacred/api/v1/Runs/1?select=captured_out,info,meta,host,experiment,artifacts,config,fail_trace,status`

Thanks!

**Comment by vivekratnavel** [comment]
That seems strange!

What is the response for `http://localhost:9000/sacred/api/v1/Runs/1` ?

**Comment by mkansy** [comment]
Also `[]`

**Comment by vivekratnavel** [comment]
Do you get a response for `http://localhost:9000/sacred/api/v1/Runs` with run id 1 ?

**Comment by mkansy** [comment]
Yes, I get a response when I enter that. It shows all of the stuff from the configuration as well as the host information (and a few other things).

If it would be helpful, I could also post it here (anonymizing my data).

**Comment by vivekratnavel** [comment]
Which version of MongoDB is this?

**Comment by mkansy** [comment]
I am using MongoDB Atlas, so I do not have MongoDB installed on my computer. It seems like it is using version 4.2 (specifically, it says 4.2.5 somewhere for the cluster).

Interestingly, I just tried `http://localhost:9000/sacred/api/v1/Runs/"1"` because I saw the 1 in quotation marks when calling `http://localhost:9000/sacred/api/v1/Runs`, and this gave me the same response for the run 1 as before (with the configuration and everything).

**Comment by vivekratnavel** [comment]
Ah, that's where the bug is. Omniboard assumes run ids to be numbers. And, that might be the issue here.

**Comment by mkansy** [comment]
Yes, I just figured out what caused the problem on my side. In my code, I create a FileStorageObserver and a MongoObserver and set it so that the MongoObserver has a lower priority (so that it takes the ID of the FileStorageObserver). Apparently, the ID of the FileStorageObserver is a string, so this is where the issue stems from.

I removed the database and now only created a MongoObserver. Now, everything works as it should again.

Thanks for your help and sorry for the mistake on my part!
